### PR TITLE
Fix convolve for zero sum kernels

### PR DESCRIPTION
**Pull request by jamienoss**
Signed-off-by: James Noss <jnoss@stsci.edu>

### CODE REVIEW

File: astropy/convolution/convolve.py [hunk 1]
```diff
@@ -219,8 +219,6 @@ def convolve(array, kernel, boundary='fill', fill_value=0.,
 
     if not kernel_sums_to_zero:
         kernel_internal /= kernel_sum
-    else:
-        kernel_internal = kernel
```
Comment (jamienoss) [inline astropy/convolution/convolve.py]: Anything that occurred before this, e.g. converting to float type ([here](https://github.com/astropy/astropy/pull/7541/files#diff-f632180d9f1e3fcf689f62e4c424e94fR175)) and filling masked kernels ([here](https://github.com/astropy/astropy/pull/7541/files#diff-f632180d9f1e3fcf689f62e4c424e94fR198)) gets undone.

**Comment by pllim** [comment]
Please add a change log for 2.0.8 under bug fixes.

@bsipocz , if backport to 2.x is not possible for this, please remilestone.

**Comment by pllim** [comment]
Failure seems unrelated but I restarted it anyway, just in case.

**Comment by pllim** [comment]
Please rebase to resolve conflicts. Thanks.

**Comment by jamienoss** [comment]
@pllim rebased

**Comment by jamienoss** [comment]
Unrelated test failure https://travis-ci.org/astropy/astropy/jobs/389300879

**Comment by bsipocz** [comment]
Thanks @jamienoss.

**Pull request merged**